test-alerts.cpp

```diff
@@ -4,5 +4,68 @@
 #include "typewise-alert.h"
 
 TEST_CASE("infers the breach according to limits") {
-  REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
+    REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
+    REQUIRE(inferBreach(32, 20, 30) == TOO_HIGH);
+    REQUIRE(inferBreach(22, 20, 30) == NORMAL);
+}
+
+TEST_CASE("classify temperate breach") {
+    REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, 25) == NORMAL);
+    REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, 36) == TOO_HIGH);
+    REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, 41) == NORMAL);
+    REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, 46) == TOO_HIGH);
+    REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, 36) == NORMAL);
+    REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, 41) == TOO_HIGH);
+}
+
+TEST_CASE("Test checkAndAlert(to email too high)") {
+
+    BatteryCharacter batterych_HAC = {
+      .coolingType = HI_ACTIVE_COOLING
+    };
+
+    std::ostringstream toEmail;
+    std::streambuf* streambuf_1 = std::cout.rdbuf();
+    std::cout.rdbuf(toEmail.rdbuf());
+
+    checkAndAlert(TO_EMAIL, batterych_HAC, 46);
+
+    //restore out stream
+    std::cout.rdbuf(streambuf_1);
+    REQUIRE(toEmail.str() == "To: a.b@c.com\nHi, the temperature is too high\n");
+}
+
+TEST_CASE("Test checkAndAlert(to email too low)") {
+
+    BatteryCharacter batterych_PC = {
+      .coolingType = PASSIVE_COOLING
+    };
+
+    std::ostringstream toEmail;
+    std::streambuf* streambuf_2 = std::cout.rdbuf();
+    std::cout.rdbuf(toEmail.rdbuf());
+
+    checkAndAlert(TO_EMAIL, batterych_PC, -10);
```

Comment:
This test is very similar to the previous one. Often, this is a signal to split it into smaller parts- how about testing the conversion from the low-classification (enum) to a human-readable message on its own? Try making that conversion as a function by itself. Then the test can just be a one-liner.